script/chart_sync_utils.sh

```diff
@@ -87,15 +109,40 @@ updateRepo() {
     sed -i.bk 's/appVersion: DEVEL/appVersion: '"${targetTagWithoutV}"'/g' "${chartYaml}"
     rm "${targetChartPath}/Chart.yaml.bk"
     # Replace images for the latest available
-    replaceImage dashboard "${targetChartPath}/values.yaml"
-    replaceImage apprepository-controller "${targetChartPath}/values.yaml"
-    replaceImage asset-syncer "${targetChartPath}/values.yaml"
-    replaceImage assetsvc "${targetChartPath}/values.yaml"
-    replaceImage kubeops "${targetChartPath}/values.yaml"
-    replaceImage pinniped-proxy "${targetChartPath}/values.yaml"
+    replaceImage_latestToProduction dashboard "${targetChartPath}/values.yaml"
+    replaceImage_latestToProduction apprepository-controller "${targetChartPath}/values.yaml"
+    replaceImage_latestToProduction asset-syncer "${targetChartPath}/values.yaml"
+    replaceImage_latestToProduction assetsvc "${targetChartPath}/values.yaml"
+    replaceImage_latestToProduction kubeops "${targetChartPath}/values.yaml"
+    replaceImage_latestToProduction pinniped-proxy "${targetChartPath}/values.yaml"
+}
+
+updateRepoWithRemoteChanges() {
+    local targetRepo=${1:?}
+    local targetTag=${2:?}
+    local targetTagWithoutV=${targetTag#v}
+    local targetChartPath="${targetRepo}/${CHART_REPO_PATH}"
+    local remoteChartYaml="${targetChartPath}/Chart.yaml"
+    local localChartYaml="${KUBEAPPS_CHART_DIR}/Chart.yaml"
+    if [ ! -f "${remoteChartYaml}" ]; then
+        echo "Wrong repo path. You should provide the root of the repository" > /dev/stderr
+        return 1
+    fi
+    rm -rf "${KUBEAPPS_CHART_DIR}"
```

Comment:
As mentioned above, not sure how this can work, since we lose any changes that we've made but not yet released.

Reply:
Not sure about it, could you elaborate  I mean, a simple example in which you think this situation might happen.